pkg/monitors/heroku/monitor.go

```diff
@@ -0,0 +1,64 @@
+package heroku
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/signalfx/signalfx-agent/pkg/core/config"
+	"github.com/signalfx/signalfx-agent/pkg/monitors"
+	"github.com/signalfx/signalfx-agent/pkg/monitors/types"
+	"github.com/signalfx/signalfx-agent/pkg/utils"
+	log "github.com/sirupsen/logrus"
+)
+
+// Config for this monitor
+type Config struct {
+	config.MonitorConfig `yaml:",inline" acceptsEndpoints:"true"`
+}
+
+// Monitor for Hereoku metadata
+type Monitor struct {
+	Output types.Output
+	cancel context.CancelFunc
+	ctx    context.Context
+	logger *utils.ThrottledLogger
+}
+
+func init() {
+	monitors.Register(&monitorMetadata, func() interface{} { return &Monitor{} }, &Config{})
+}
+
+// Configure monitor
+func (m *Monitor) Configure(c *Config) error {
+	m.logger = utils.NewThrottledLogger(log.WithFields(log.Fields{"monitorType": "heroku-metadata"}), 20*time.Second)
+	m.ctx, m.cancel = context.WithCancel(context.Background())
+
+	utils.RunOnInterval(m.ctx, func() {
```

Comment:
This doesn't need to run on an interval, only once.

Reply:
But do the run in a goroutine.

Reply:
Updated